runtime/interpreter/interpreter.go

```diff
@@ -1187,14 +1188,20 @@ func (declarationInterpreter *Interpreter) declareNonEnumCompositeValue(
 
 	var initializerFunction FunctionValue
 	if declaration.Kind() == common.CompositeKindEvent {
-		initializerFunction = NewHostFunctionValue(
+		// Initializer is a static function.
+		initializerFunction = NewStaticHostFunctionValue(
```

Comment:
Should this be a bound function?

Reply:
hmm, I initially thought this is same as the constructor. But seems not? In that case, yeah, this can be a bound function.